pyasn1/codec/ber/decoder.py

```diff
@@ -16,35 +19,71 @@
 from pyasn1.type import univ
 from pyasn1.type import useful
 
-__all__ = ['decode']
+
+__all__ = ['decode', 'decodeStream']
 
 LOG = debug.registerLoggee(__name__, flags=debug.DEBUG_DECODER)
 
 noValue = base.noValue
 
 
+def asStream(substrate):
+    # TODO: Apply sparingly or remove
+    # :type substrate: Union[bytes, BytesIO, BufferedReader, OctetString]
+    # :rtype: BufferedReader
+    if isinstance(substrate, univ.OctetString):
+        substrate = substrate.asOctets()
+    if isinstance(substrate, bytes):
+        substrate = BytesIO(substrate)
+    if isinstance(substrate, BytesIO):
+        return BufferedReader(substrate)
+    if isinstance(substrate, BufferedReader):
+        return substrate
+    raise ValueError("%s Cannot be converted to stream" % substrate.__class__)
```

Comment:
Silly question: can it be just a file descriptor or file-like object? The use-case would be to process running stream from stdin.

Reply:
We should have that. I agree.